### PR TITLE
Add External network attachment support for vcd_nsxt_edgegateway

### PR DESCRIPTION
**Pull request by Didainius**
This PR adds support for adding additional External Networks (only NSX-T Segment backed) to `vcd_nsxt_edgegateway`. This is done using a block field `external_network`. It is in the same resource instead of a separate one because it uses the same API endpoint as Edge Gateway creation (a bit unfortunate, but safest way). 

This feature requires VCD 10.4.1+ and uses new functionality in https://github.com/vmware/go-vcloud-director/pull/610

### Place in UI:
image

### About
IP Management for Edge Gateway external network uplinks is quite simple - it doesn't support IP Spaces and only requires IP, Prefix and allocated IP count. Primary IP can either be specified manually or assigned automatically. `TypeSet` has enough functionality to support it therefore only one structure is introduced `external_network`.

A big part of the work was making sure that this works well with existing primary network uplinks and doesn't affect each other. Quite a few tests were added with the hope of ensuring interoperability.

### Additional notes:
* Attempt to improve `TestAccVcdNsxtDynamicSecurityGroupVdcGroupCriteriaWithVms` where it sometimes fails with incorrect VM count
* Added missing changelog note for #1092
* Test config adds new field `testConfig.Nsxt.NsxtImportSegment2` for improved testing

### Testing
Tags used `gateway nsxt network`, VCD versions:
* 10.4.1
* 10.5.0
* 10.4.0
* 10.4.2

### CODE REVIEW

File: .changes/v3.11.0/1092-improvements.md [hunk 1]
```diff
@@ -0,0 +1,3 @@
+* Resources `vcd_vapp_network` and `vcd_vapp_org_network` will additionally check if vApp is in
```
Comment (Didainius) [inline .changes/v3.11.0/1092-improvements.md:1]: This is covering other PR #1092

File: website/docs/r/nsxt_edgegateway.html.markdown [hunk 2]
```diff
@@ -307,6 +347,11 @@ can be used to lookup ID by name.
   backed by *IP Spaces*. Read more in [IP allocation modes](#ip-allocation-modes) section.
 * `total_allocated_ip_count` - (Optional, *v3.9+*) Required with `subnet_with_total_ip_count`. It is
   **read-only** attribute with other other allocation models `subnet` and `subnet_with_ip_count`.
+  **Note**. It sets or reports IP count *only for NSX-T Tier 0 backed External Network Uplink*.
+* `external_network` - (Optional,*VCD 10.4.1+*, *v3.11+*) attaches NSX-T Segment backed External Networks with a
```
Comment (lvirbalas) [inline website/docs/r/nsxt_edgegateway.html.markdown:351]:
```suggestion
* `external_network` - (Optional, *VCD 10.4.1+*, *v3.11+*) attaches NSX-T Segment backed External Networks with a
```

File: website/docs/r/nsxt_edgegateway.html.markdown [hunk 3]
```diff
@@ -374,15 +419,31 @@ added capability of automatically allocating a specific number of IPs from the b
...
 * `allocated_ip_count` (Required) - Number of allocated IPs from that particular subnet
 
+<a id="edgegateway-subnet-external-network"></a>
+
+### NSX-T Segment backed network attachment using `external_network` block
+
+This option can attach NSX-T Segment backed External Networks
+
+* `external_network_id` - (Required) - ID of NSX-T Segment backed External Network
+* `gateway` - (Required) - Gateway for a subnet in external network
```
Comment (lvirbalas) [inline website/docs/r/nsxt_edgegateway.html.markdown:429]: Please chose one of "External Network" or "external network" in this bullet list for consistency :)

Comment (Didainius) [reply website/docs/r/nsxt_edgegateway.html.markdown:429]: It was 9 occurrences for lower cases vs. 8 for upper cases in this document. I made them all lower cased now :)

File: website/docs/r/nsxt_edgegateway.html.markdown (hunk 2 above)
Comment (Didainius) [reply website/docs/r/nsxt_edgegateway.html.markdown]: Fixed.

**Review by lvirbalas — APPROVED** [review]
Thanks for updates!

**Review by dataclouder — APPROVED** [review]

File: .changes/v3.11.0/1092-improvements.md [hunk 4]
```diff
@@ -0,0 +1,3 @@
...
+  `RESOLVED` (in addition to already checked `POWERED_OFF`) state before attempt a reboot when
```
[the remark this answers is not in the document]
Comment (Didainius) [reply .changes/v3.11.0/1092-improvements.md]: Fixed.

**Review by adambarreiro — APPROVED** [review]

File: vcd/ipv6_test.go [hunk 5]
```diff
@@ -439,3 +475,53 @@ resource "vcd_nsxt_ip_set" "ipv6" {
...
+      start_address = "2002:0:0:1234:abcd:ffff:c0a6:122"
+      end_address   = "2002:0:0:1234:abcd:ffff:c0a6:124"
+    }
+  }
+}
+
+resource "vcd_nsxt_edgegateway" "nsxt-edge" {
+  org         = "{{.Org}}"
+  owner_id    = vcd_org_vdc.with-edge-cluster.id
+  name        = "{{.TestName}}-edge-gateway"
```
Comment (adezxc) [inline vcd/ipv6_test.go:505]: nit, indentation is a bit off

**Review by adezxc — APPROVED** [review]

File: vcd/ipv6_test.go (hunk 5 above)
Comment (Didainius) [reply vcd/ipv6_test.go]: Fixed.

**Pull request merged**